source/javascripts/components/unified-editor/WorkflowConfig/WorkflowConfigPanel.tsx

```diff
@@ -33,23 +36,26 @@ const WorkflowConfigPanelContent = () => {
   });
 
   useEffect(() => {
-    if (originalName) {
+    if (defaultWorkflowId) {
       const { configuration, properties } = formValues;
 
       if (configuration) {
-        const { stackId = '', machineTypeId = '' } = configuration;
-        updateStackAndMachine(originalName, stackId, machineTypeId);
+        const { stackId = '', machineTypeId = '', envs = [] } = configuration;
+        const ymlCompatibleEnvVars = envs.map((env) => EnvVarService.parseEnvVar(env as EnvVar));
```

Comment:
I am wondering if it would make sense to push this conversation down to the BitriseYMLStore? E.g. for stack and machines, we don't expect clients to know that stack and machine are under the "bitrise-io" field.